inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp

```diff
@@ -0,0 +1,1950 @@
+    scales.resize(scalesLen);
+    for (int i = 0; i < scalesLen; i++) {
+        scales[i] = scalesData[i];
+    }
+
+    SizeVector srcDim = getParentEdgeAt(DATA_ID)->getDesc().getDims();
+    size_t dimSize = srcDim.size();
+    if (isAxesSpecified) {
+        auto &axesMemPtr = getParentEdgeAt(AXES_ID)->getMemoryPtr();
+        int *axesData = reinterpret_cast<int*>(axesMemPtr->GetData()) +
+            axesMemPtr->GetDescriptor().data.layout_desc.blocking.offset_padding;
+        int axesLen = getParentEdgeAt(AXES_ID)->getDesc().getDims()[0];
+        axes.resize(axesLen);
+        for (int i = 0; i < axesLen; i++) {
+            axes[i] = axesData[i];
+        }
+    } else {
+        // default is: [0, ..., dataRank - 1]
+        axes.resize(dimSize);
+        for (int i = 0; i < dimSize; i++) {
+            axes[i] = i;
+        }
+    }
+    if (scales.size() != axes.size() || targetShape.size() != axes.size()) {
+        THROW_IE_EXCEPTION << "Interpolate layer with name '" << getName() <<
+        "' do not have the same number elements in axis as in scales or target_spatial_shape.";
+    }
+
+    SizeVector srcDimPad = getPaddedInputShape(srcDim);
+    SizeVector dstDim = outShapeCalc(srcDim);
+
+    auto to5Dim = [&](SizeVector casesDim) {
+        size_t caseSize = casesDim.size();
+        SizeVector dim5(5, 1lu);
+        if (caseSize > 2) {
+            dim5[0] = casesDim[0];
+            dim5[1] = casesDim[1];
+        }
+        if (caseSize == 5) {
+            dim5[2] = casesDim[2];
+        }
+        dim5[3] = casesDim[caseSize - 2];
+        dim5[4] = casesDim[caseSize - 1];
+        return dim5;
+    };
+    auto srcDim5d = to5Dim(srcDim);
+    auto srcDimPad5d = to5Dim(srcDimPad);
+    auto dstDim5d = to5Dim(dstDim);
+
+    uint8_t *src_data = nullptr;
+    std::vector<uint8_t> srcPadded;
+    if (hasPad) {
+        int padB0 = (dimSize > 2) ? padBegin[0] : 0;
+        int padB1 = (dimSize > 2) ? padBegin[1] : 0;
+        int padB2 = (dimSize == 5) ? padBegin[dimSize - 3] : 0;
+        int padB3 = padBegin[dimSize - 2];
+        int padB4 = padBegin[dimSize - 1];
+
+        SizeVector inShapeBlock = getBlockND(srcDim5d);
+        SizeVector inShapePadBlock = getBlockND(srcDimPad5d);
+        srcPadded = std::vector<uint8_t>(inShapePadBlock[0] * srcDataSize, 0);
+        uint8_t *src_data_pad = static_cast<uint8_t *>(&srcPadded[0]);
+
+        parallel_for4d(srcDim5d[0], srcDim5d[1], srcDim5d[2], srcDim5d[3], [&](int n, int c, int d, int h) {
+            uint8_t *src = src_data_origin + (inShapeBlock[1] * n + inShapeBlock[2] * c + inShapeBlock[3] * d + inShapeBlock[4] * h) * srcDataSize;
+            uint8_t *srcPad = src_data_pad + (inShapePadBlock[1] * (n + padB0) + inShapePadBlock[2] * (c + padB1) +
+                           inShapePadBlock[3] * (d + padB2) + inShapePadBlock[4] * (h + padB3) + padB4) * srcDataSize;
+            std::memcpy(srcPad, src, srcDim5d[4] * srcDataSize);
```

Comment:
In general I don't like this approach. Interpolate in most cases is memory bound operation. That's whu full coping of src data my increase inference time up to 2x. Since copying is done only in case `hasPad == true` I am ok to keep it for now. However we need to consider this for future optimizations.

Reply:
IMO, hasPad is a rare case, so I copied. I will do this with in-place manner if there is a case need it in the future.